kmod/src/ioctl.c

```diff
@@ -937,6 +938,55 @@ static long scoutfs_ioc_alloc_detail(struct file *file, unsigned long arg)
 	       args.copied;
 }
 
+static long scoutfs_ioc_move_blocks(struct file *file, unsigned long arg)
+{
+	struct inode *to = file_inode(file);
+	struct super_block *sb = to->i_sb;
+	struct scoutfs_ioctl_move_blocks __user *umb = (void __user *)arg;
+	struct scoutfs_ioctl_move_blocks mb;
+	struct file *from_file;
+	struct inode *from;
+	int ret;
+
+	if (copy_from_user(&mb, umb, sizeof(mb)))
+		return -EFAULT;
+
+	if (mb.len == 0)
+		return 0;
+
+	if (mb.from_off + mb.len < mb.from_off ||
+	    mb.to_off + mb.len < mb.to_off)
+		return -EOVERFLOW;
+
+	from_file = fget(mb.from_fd);
+	if (!from_file)
+		return -EBADF;
+	from = file_inode(from_file);
+
+	if (from == to) {
+		ret = -EINVAL;
+		goto out;
+	}
+
+	if (from->i_sb != sb) {
+		ret = -EXDEV;
+		goto out;
+	}
+
+	/* is it enough to only get one file? */
+	ret = mnt_want_write_file(file);
```

Comment:
dunno, looks like you could call `mnt_clone_write()` here if you wanted (and then drop_write() twice) but what would be the point?

Reply:
Huh, I hadn't noticed that call.  But now that I've dug in, I don't think we need more than one.  I was worried that the _file() variant actually did something to the file, but it doesn't, it just uses it to get at the file's mount point.  So one call is enough 'cause we specifically require that the two files be on one mount.  I'll remove the comment.